libraries/lib-utility/AppEvents.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ * SPDX-FileName: AppEvents.h
+ * SPDX-FileContributor: Dmitry Vedenko
+ */
+
+#pragma once
+
+#include <functional>
+
+namespace AppEvents
+{
+/*! Register callback to be called when application is initialized.
+ *  If application is already initialized, callback will be called immediately.
+ *  @param callback Callback to be called when application is initialized.
+ *  @pre `!!calback`
+ */
+UTILITY_API void OnAppInitialized(std::function<void()> callback);
+
+class UTILITY_API AppEventsProvider /* not final */
```

Comment:
My guess is that `AppEventsProvider` exists to protect parts of events API from unwanted use? If so I'd suggest also to make a statement to the class comment that the number of instances should be limited to one (assertion in ctor wouldn't harm).
`AppEvents::AppEventsProviders` could be shortened to `AppEvents::Provider`, sorry :)